### PR TITLE
Posts page feature

### PR DESCRIPTION
**Pull request by George-Hachem**
Posts are generated in the feed from newest to oldest. I also created a modal overlay for each post, which contains a description tab and a comment tab as well. In the description, there is a message and like button. The message button will create a conversation between the two users. The like button functions like the Instagram like button; if you press it for the first time, the number of likes increments and if you press is again it will decrease. I also added small details like owners of t posts will see a delete button on the top right of the post, they won't be able to like their own posts(because the like button is like a save button and there is no reason for owners to save their own posts, the posts will already be projected in their profiles), and the message button won't show on their own posts(since it does not make sense for users to message themselves).

### CODE REVIEW

**Comment by IanMcC123** [comment]
Great start George. Great demo. I think that there is room for potential future styling, but looks good for now.

**Review by IanMcC123 — APPROVED** [review]

File: frontend/src/pages/messaging/messaging.jsx [hunk 1]
```diff
@@ -144,6 +150,41 @@ export default function Messaging() {
         setSearchUsername('');
         setSearchResult(null);
     };
+    // const handleStartChatButtonClick = async (newPostUserId = null) => {
+    //     const userId = newPostUserId ?? selectedUser[0]._id;
```
Comment (shihab3750) [inline frontend/src/pages/messaging/messaging.jsx:154]: Can you remove this

File: frontend/src/pages/messaging/messaging.jsx [hunk 2]
```diff
@@ -188,13 +229,16 @@ export default function Messaging() {
...
-    }
+    // useEffect(() => {
+    //     if (postUserId && userData) {
+    //         /*
+    //             If valued
+    //                 Select conversation where user ids are equal to postUserId & userData._id
+    //             If not valued
+    //                 Don't do anything
+    //         */
+    //     }
```
Comment (shihab3750) [inline frontend/src/pages/messaging/messaging.jsx:240]: not being used

**Review by shihab3750 — CHANGES_REQUESTED** [review]
clean up needed

**Review by shihab3750 — APPROVED** [review]
Approved

**Pull request merged**